frameworks/Rust/xitca-web/src/db.rs

```diff
@@ -0,0 +1,146 @@
+use std::{cell::RefCell, error::Error, fmt::Write};
+
+use ahash::AHashMap;
+use futures_util::stream::{FuturesUnordered, TryStreamExt};
+use rand::{rngs::SmallRng, Rng, SeedableRng};
+use tokio::pin;
+use tokio_postgres::{types::ToSql, NoTls, Statement};
+
+use super::ser::{Fortune, Fortunes, World};
+
+pub struct Client {
+    client: tokio_postgres::Client,
+    rng: RefCell<SmallRng>,
+    fortune: Statement,
+    world: Statement,
+    updates: AHashMap<u16, Statement>,
+}
+
+pub async fn create(config: &str) -> Client {
+    let (client, conn) = tokio_postgres::connect(config, NoTls).await.unwrap();
+
+    tokio::task::spawn_local(async move {
+        let _ = conn.await;
+    });
+
+    let fortune = client.prepare("SELECT * FROM fortune").await.unwrap();
+
+    let mut updates = AHashMap::new();
+    for num in 1..=500u16 {
+        let mut pl = 1;
+        let mut q = String::new();
+        q.push_str("UPDATE world SET randomnumber = CASE id ");
+        for _ in 1..=num {
+            let _ = write!(&mut q, "when ${} then ${} ", pl, pl + 1);
+            pl += 2;
+        }
+        q.push_str("ELSE randomnumber END WHERE id IN (");
+        for _ in 1..=num {
+            let _ = write!(&mut q, "${},", pl);
+            pl += 1;
+        }
+        q.pop();
+        q.push(')');
+
+        let st = client.prepare(&q).await.unwrap();
+        updates.insert(num, st);
+    }
```

Comment:
Is this really creating 500 prepared update statements?

Reply:
Yes I believe it does prepare all possible update query and store them in client object. I copied this piece of code from actix bench and made slight change to make it compile.

Reply:
Ah, now I see that xitca is actix backwards...